data/migrations/versions/e9f3e4dbb979_quota_management_reporting.py

```diff
@@ -0,0 +1,93 @@
+"""quota_management_reporting
+
+Revision ID: e9f3e4dbb979
+Revises: 909d725887d3
+Create Date: 2021-12-16 15:14:43.054705
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'e9f3e4dbb979'
+down_revision = '909d725887d3'
+
+import sqlalchemy as sa
+
+
+def upgrade(op, tables, tester):
+
+    op.create_table(
+        "quotalimitgroups",
+        sa.Column("id", sa.Integer, nullable=False),
+        sa.Column("group_name", sa.String(length=255), nullable=False),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_quotalimitgroups"))
+    )
+
+    op.create_table(
+        "userorganizationquota",
+        sa.Column("id", sa.Integer, nullable=False),
+        sa.Column("organization_id", sa.Integer, nullable=False),
+        sa.Column("limit_bytes", sa.Integer, nullable=False),
+        sa.Column("quota_limit_group_id", sa.Integer, nullable=False),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_userorganizationquota")),
+        sa.ForeignKeyConstraint(
+            ["quota_limit_group_id"], ["quotalimitgroups.id"], name=op.f("fk_userorganizationquota_limit_group")
+        ),
+        sa.ForeignKeyConstraint(
+            ["organization_id"], ["user.id"], name=op.f("fk_userorganizationquota_organization")
+        )
+    )
+
+    op.create_index(
+        "userorganizationquota_organization", "userorganizationquota", ["organization_id"], unique=True
+    )
+    op.create_index(
+        "userorganizationquota_limitgroup", "userorganizationquota", ["quota_limit_group_id"], unique=False
+    )
+
+    op.create_table(
+        "quotatype",
+        sa.Column("id", sa.Integer, nullable=False),
+        sa.Column("description", sa.String(length=255), nullable=False),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk_quotatype"))
+    )
+
+
+    op.create_table(
+        "quotalimits",
+        sa.Column("id", sa.Integer, nullable=False),
+        sa.Column("quota_limit_group_id", sa.Integer, nullable=False),
+        sa.Column("quota_type_id", sa.Integer, nullable=False),
+        sa.Column("percent_of_limit", sa.Integer, nullable=False),
+        sa.ForeignKeyConstraint(["quota_type_id"], ["quotatype.id"], name=op.f("fk_quotalimit_type")),
+        sa.ForeignKeyConstraint(["quota_limit_group_id"], ["quotalimitgroups.id"], name=op.f("fk_quotalimit_limit_group"))
+    )
+
+    op.create_index(
+        "quotalimits_limitgroupid", "quotalimits", ["quota_limit_group_id"], unique=False
+    )
+
+    op.create_table(
```

Comment:
Can we split this work into multiple PRs: first the reporting part (i.e anything that deals with the calculations of repo size/limits) and the actual soft//hard quota part.

Any quota implementation will need a working reporting api first anyways, and this will make it easier to track/review.

Reply:
Ideally the first PR would only include the repositorysize table, since that's the only one needed for the reporting implementation.

Reply:
Ideally for the reporting portion we would be able to report how much is in use but also how much is "allowed" and how much of the quota that has been set is being used. I think it best to keep the entire schema and provide the reporting across all of the features. The code in v2 push is currently a placeholder for a quick easy way to learn and execute the code, I will add it to the v1 reporting endpoints soon.